state.go

```diff
@@ -27,76 +30,848 @@ type State struct {
 	InterimTranscriptHash   []byte
 
 	// Shared secret state
-	Keys KeyScheduleEpoch
+	Keys keyScheduleEpoch
 
 	// Per-participant state
 	Index        leafIndex
 	IdentityPriv SignaturePrivateKey
+	scheme       SignatureScheme
 
 	// Cache of proposals and update secrets
 	// XXX: The map key for UpdateSecrets should actually be ProposalID, but that
 	// struct can't be used as a map key because it's not comparable (because
 	// slices are not comparable).  Instead, we use ProposalID.String()
 	PendingProposals []MLSPlaintext
 	UpdateSecrets    map[string][]byte
+
+	// stored groupContext
+	grpContext GroupContext
```

Comment:
You can probably just cache this as a []byte, right?  You never need to access the constituent components.